lib/Interpreter/InterOp.cpp

```diff
@@ -2184,5 +2184,25 @@ namespace InterOp {
     }
     return dims;
   }
+
+  bool IsTypeDerivedFrom(TCppType_t derived, TCppType_t base)
+  {
+    auto derivedType = clang::QualType::getFromOpaquePtr(derived);
+    auto baseType = clang::QualType::getFromOpaquePtr(base);
+
+    if (derivedType.isNull() || baseType.isNull())
+      return false;
+
+    auto derivedDecl = derivedType->getAsCXXRecordDecl();
+    auto baseDecl = baseType->getAsCXXRecordDecl();
+
+    if (!derivedDecl || !baseDecl)
+      return false;
+
+    if (derivedDecl == baseDecl)
+      return true;
+
+    return derivedDecl->isDerivedFrom(baseDecl);
+  }
```

Comment:
```suggestion
  bool IsTypeDerivedFrom(SemaType *S, TCppType_t derived, TCppType_t base)
  {
    auto derivedType = clang::QualType::getFromOpaquePtr(derived);
    auto baseType = clang::QualType::getFromOpaquePtr(base);
    return S->isDerivedFrom(derivedType, baseType);
  }
```

I was suggesting something like that.